### ISSUE TITLE
`composer req admin` fails with "There is no extension able to load the configuration for "doctrine"

### ISSUE THREAD

**Issue by guzzlefry** [symfony/flex#82]
I've tried running this immediately after `composer clear-cache` with no success.

The exact command I'm running is:
`composer -v create-project symfony/skeleton:3.3.x-dev demo && cd demo && composer req webserver && composer req cli && composer req admin && ./bin/console server:start`

### `php -v` output:
```
PHP 7.1.4-1+deb.sury.org~xenial+1 (cli) (built: Apr 11 2017 22:12:32) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.1.4-1+deb.sury.org~xenial+1, Copyright (c) 1999-2017, by Zend Technologies`
```

### `composer -V` output:
```
Composer version 1.4.1 2017-03-10 09:29:45
```

### Error:
```
Executing script assets:install --symlink --relative %WEB_DIR% [KO]
 [KO]
Script assets:install --symlink --relative %WEB_DIR% returned with error code 1
!!  
!!  
!!                                                                                 
!!    [Symfony\Component\Config\Exception\FileLoaderLoadException]                 
!!    There is no extension able to load the configuration for "doctrine" (in /ho  
!!    me/gus/Projects/Learning/demo/etc/packages/doctrine.yaml). Looked for names  
!!    pace "doctrine", found "framework", "web_server", "easy_admin" in /home/gus  
!!  
!!    /Projects/Learning/demo/etc/packages/doctrine.yaml (which is loaded in reso  
!!    urce "/home/gus/Projects/Learning/demo/etc/packages/doctrine.yaml").         
!!                                                                                 
!!  
!!  
!!                                                                                 
!!    [Symfony\Component\DependencyInjection\Exception\InvalidArgumentException]   
!!    There is no extension able to load the configuration for "doctrine" (in /ho  
!!    me/gus/Projects/Learning/demo/etc/packages/doctrine.yaml). Looked for names  
!!    pace "doctrine", found "framework", "web_server", "easy_admin"               
!!  
!!                                                                                 
!!  
!!  
!!  

Installation failed, reverting ./composer.json to its original content.
```

### Full console output is provided below:
[flex_debug.txt](https://github.com/symfony/flex/files/994671/flex_debug.txt)

**Comment by fabpot** [comment]
That's really weird as I've just tried the exact same command and it works for me.

**Comment by fabpot** [comment]
Can you create a zip of the project's directory and send it to me for further investigations?

**Comment by guzzlefry** [comment]
Sure: [demo.zip](https://github.com/symfony/flex/files/994754/demo.zip)
I'll also try running against some VMs/Docker containers to see if the issue is present there.

**Comment by javiereguiluz** [comment]
@guzzlefry I've executed the command you gave and it worked for me. Then I downloaded your ZIP and made a recursive `diff` between your project and mine. Besides the obvious differences in logs and cache, these are the things that differ between the two projects:

```diff
diff -r demo/composer.json demo2/composer.json
6a7
>         "javiereguiluz/easyadmin-bundle": "^1.16@dev",
diff -r demo/etc/bundles.php demo2/etc/bundles.php
5a6,10
>     'Symfony\Bundle\TwigBundle\TwigBundle' => ['all' => true],
>     'Symfony\Bundle\SecurityBundle\SecurityBundle' => ['all' => true],
>     'Sensio\Bundle\FrameworkExtraBundle\SensioFrameworkExtraBundle' => ['all' => true],
>     'Doctrine\Bundle\DoctrineCacheBundle\DoctrineCacheBundle' => ['all' => true],
>     'Doctrine\Bundle\DoctrineBundle\DoctrineBundle' => ['all' => true],
```

All those lines are present in my project but not in yours.

**Comment by guzzlefry** [comment]
I suspect those are added if the `composer req admin` call actually succeeds.

Also, I run into this same issue on a new user on the same system. I'm compiling PHP on another box to see if the issue is also present there.

**Comment by stof** [comment]
@fabpot it would be great if Flex could display debugging info about whether it found recipes on the server, whether it autocreated a bundle recipe or whether no recipe was found at all for the package (displaying this info only in very verbose mode or in the debug verbosity). This would make it much easier to figure out why DoctrineBundle is not registered properly, by asking people to increase the verbosity of the composer command.
Btw, this would be consistent with the behavior of Composer, which takes full advantage of the different verbosity levels of symfony/console

**Comment by robinlehrmann** [comment]
I think the problem is the opcache.
The `load` method in the `BundlesConfigurator` requires the bundles.php
The following line can fix it:
```
private function load(string $file): iterable
    {
        if (function_exists('opcache_invalidate')) {
            opcache_invalidate($file, true);
        }
        $bundles = file_exists($file) ? (require $file) : [];
        if (!is_array($bundles)) {
            $bundles = [];
        }

        return $bundles;
    }
```

**Comment by stof** [comment]
the right place to invalidate is actually in the code *writing* to this file

**Comment by robfrawley** [comment]
I _also_ got this error a few days ago but hadn't gotten around to filing an issue for it.

**Comment by guzzlefry** [comment]
I forgot to mention that I'm on Ubuntu 16.04.

I've confirmed that disabling the opcache extension fixes the issue, while re-enabling causes the issue to reappear.

Here's a dump of `php -i` sans environmental variables and PHP variables. I can provide those too if needed, just not sure if anything confidential would be stored in those.
[php_info.txt](https://github.com/symfony/flex/files/998100/php_info.txt)

**Comment by PeterFour** [comment]
I'm having the same error, not the same require but same end result in the end.
I can paste my terminal log if needed, or the project dir

**Comment by xabbuh** [comment]
@PeterFour Can you confirm that this is related to Opcache for you too?

**Comment by PeterFour** [comment]
Humm looks like after doing a `phpdismod -s cli opcache` I still have the same error when I do `composer req admin`.

However wiping the project and starting again worked; my bad I should have read the comment more thoroughly.

**Comment by xabbuh** [comment]
please try the fix from #84
